### PR TITLE
Add azure.developer.* namespace to registered namespaces

### PR DESCRIPTION
**Pull request by catalinaperalta**
Adding the azure.developer.* namespace for services like Dev Center and Load Testing.

### CODE REVIEW

**Comment by catalinaperalta** [comment]
Link to previous arch board reviews where this namespace was discussed: https://github.com/Azure/azure-sdk/issues/4702, https://github.com/Azure/azure-sdk/issues/6824 + [recording link](https://microsoft-my.sharepoint.com/:v:/p/teglaza/EUrrb7ieES1Pi_Ukvr0W7yUBNzOgRLveMMAD6db_wGEN3A?referrer=Teams.TEAMS-ELECTRON&referrerScenario=MeetingChicletGetLink.view.view)

**Comment by catalinaperalta** [comment]
@KrzysztofCwalina @tg-msft @JonathanGiles @annatisch @bterlson @srnagar @johanste could we get the `azure.developer.*` namespace approved? See my previous comment about the packages that currently fall under this namespace. Thanks

**Comment by JonathanGiles** [comment]
No issue here for Java - go for it! Thanks.

File: docs/tables/registered_namespaces.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ The following are a list of registered namespaces.
 | `azure.ai.formrecognizer`     | [Form Recognizer]              |
 | `azure.ai.textanalytics`      | [Text Analytics]               |
 | `azure.data.appconfiguration` | [App Configuration]            |
+| `azure.developer.devcenter`   | [Dev Center]                   |
 | `azure.cosmos`                | [Azure Cosmos DB]              |
 | `azure.messaging.eventhubs`   | [Event Hubs]                   |
```
Comment (tg-msft) [inline docs/tables/registered_namespaces.md:17]: Can we fix the previously broken sort here as part of this change too?

```suggestion
| `azure.ai.textanalytics`      | [Text Analytics]               |
| `azure.cosmos`                | [Azure Cosmos DB]              |
| `azure.data.appconfiguration` | [App Configuration]            |
| `azure.developer.devcenter`   | [Dev Center]                   |
| `azure.messaging.eventhubs`   | [Event Hubs]                   |
```

**Review by tg-msft — DISMISSED** [review]

**Review by johanste — DISMISSED** [review]

**Review by weshaggard — APPROVED** [review]

**Pull request merged**